mage/src/main/java/mil/nga/giat/mage/data/repository/user/UserRepository.kt

```diff
@@ -308,15 +308,20 @@ class UserRepository @Inject constructor(
    private suspend fun syncIcon(user: User) {
       val path = "${MediaUtility.getUserIconDirectory(application)}/${user.id}.png"
       val file = File(path)
-      if (file.exists()) {
+      /// At the moment, I can't see any code that regularly updates User.lastModified
```

Comment:
For reference, lastUpdated is tracked by the server and updated when the application fetches a user.

https://github.com/ngageoint/mage-android/blob/master/mage/src/main/java/mil/nga/giat/mage/network/user/UserWithRoleTypeAdapter.kt#L52-L55
and
https://github.com/ngageoint/mage-android/blob/master/mage/src/main/java/mil/nga/giat/mage/network/user/UserWithRoleIdTypeAdapter.kt#L48-L51

Users should only be fetched when selecting an event.  This flow happens after login when selecting an event or after changing events.

Reply:
My comment wasn’t meant to suggest that your change was fundamentally wrong. Rather, it was intended to provide some context, if helpful, regarding the added comments.

For example, you mention:
> the fact that we happen to pull users in their entirety every time we do something...

Users _should_ currently be pulled when an event is selected, either at login or when changing events after login. Is that not the case? This would be drastically different from pulling them every time we do something.

The client has no way of determining if a user's remote properties have changed without retrieving that information. While we do have local storage for the user, it could be stale.